### PR TITLE
[PROTOCOL-NG] Solidity Semantic Money draft release 1

### PR DESCRIPTION
**Pull request by hellwolf**
## Background

This is the first release of the solidity semantic money library, used for building GDA in SuperToken v1 and SuperToken v2. The audience of this is the implementers of SuperTokens on any target system.

:warning: Note that many concepts are brand new without prio-art, and the time to provide the detailed documentation for the new idea is limited. As a result, considering the fairly limited audience for this library, for ones actually interested in using it, instead of asking for the documentation, one should contact the author and Superfluid team for further assistance to your work.

## What's In the Bag

- A Solidity library for building the general payment primitives aspect of the semantic money is ready:
  - A pure functional solidity library: `SemanticMoney`.
    - Property based testing using foundry fuzz.
    - (Limited, WIP) Certora formal verification using CVL (Certora Verification Language) v1.
  - A token monad using abstract contract approach: `TokenMonad.
  - Toy models for `ToySuperToken` & `TokenSuperfluidPool`, as reference implementation for GDA & SuperToken v2.
    - Property based testing using foundry fuzz, notably also with a randomized steps generation testing in it (to be replaced by foundy invariance testing in the future).
  - Toy model for Aqueduct, a ZILMM (Zero Intermediate Liquidity  Money Market).
    - Property based testing using foundry fuzz.

- ~Update the yellow paper~ deferred to next PR.

### CODE REVIEW

File: packages/solidity-semantic-money/src/ref-impl/ISuperfluidPool.sol [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    function getDisconnectedFlowRate() external view returns (FlowRate);
+
+    /**
+     * @dev Disconnected balance, or all disconnected members' claimable balances.
+     *
+     * Property:
+     *
+     *   (a) pool.getDisonnectedBalance(t) == foldr sum (\m -> m.getClaimable(m, t)) 0 (isDisconnectd pool.members)
+     */
+    function getDisonnectedBalance(Time t) external view returns (Value);
```
Comment (0xdavinchee) [inline packages/solidity-semantic-money/src/ref-impl/ISuperfluidPool.sol]: minor typo here and two other places

Comment (hellwolf) [reply packages/solidity-semantic-money/src/ref-impl/ISuperfluidPool.sol]: Thanks, addressed.

**Comment by hellwolf** [comment]
If this got merged and used by the #1357 branch, let's merge so that GDA branch will be cleaner to maintain. I will do the a series of library explanation sessions as a separate and an ongoing process live.

**Review by 0xdavinchee — APPROVED** [review]
LGTM

**Pull request merged**